containers/Dockerfile.base-spack

```diff
@@ -59,37 +59,31 @@ ARG SPACK_REPO_VERSION=releases/${SPACK_VERSION}
 ARG SPACK_PACKAGES_REPO_VERSION=main
 ARG SPACK_CONFIG_REPO_VERSION=main
 ARG SPACK_ARCH=linux-rocky8-x86_64
+ARG SPACK_CONFIG_DIR=/opt/spack-config/${SPACK_VERSION}/ci
 ARG COMPILER_PKG_NAME=intel-oneapi-compilers
 ARG COMPILER_PKG_VERSION=2021.2.0
 ARG COMPILER_NAME=intel
 ARG COMPILER_VERSION=2021.2.0
 
 ENV SPACK_ROOT=/opt/spack
 ENV GNUPGHOME=${SPACK_ROOT}/opt/spack/gpg
-ENV ENV_SPACK_ARCH=${SPACK_ARCH}
```

Comment:
There is a single usage of this `ENV` in the `model-1-build.yml` workflow, here: https://github.com/ACCESS-NRI/build-ci/blob/3c0840d775d8f3d67cfeedb44173caa2682fa27e/.github/workflows/model-1-build.yml#L113. We'd need to get this information from somewhere else, in that case, or remove the `arch` specifier.

Reply:
Good catch! Do you have preference?

Reply:
Good catch! Do you have a preference?

Reply:
The plan is to:
1. Try using `target` instead of `arch`.
2. Then read the `target` from `compilers.json`.

Reply:
Hi @CodeGat , Let me know when the PR to read `target` from `compilers.json`, is ready for review.

Reply:
I take it back, there is a few references to the compiler environment variables here:
https://github.com/ACCESS-NRI/build-ci/blob/3c0840d775d8f3d67cfeedb44173caa2682fa27e/containers/setup-spack-envs.sh#L10-L13
I'll see if I can get that information from the workflow I suppose.

Reply:
Changed the commits to keep the ENV variables that are needed.